INTEGRATING.md

```diff
@@ -1,31 +1,103 @@
 # Integrating alertmanager-k8s
 
+Alermanager can handle different types of relations in the `provides` side and in the `requires` side.
+
 ## Provides
 
-### alertmanager_dispatch
+### Alerting
+
+```yaml
+  alerting:
+    interface: alertmanager_dispatch
+```
 
-Any charm that implements the
+Over the the
 [`alertmanager_dispatch`](https://charmhub.io/alertmanager-k8s/libraries/alertmanager_dispatch)
-relation interface can be related to this charm for forwarding alerts to alertmanager,
+relation interface Alermanager can be related to charms that can forward alerts to it,
 for example: [Prometheus][Prometheus operator], [Loki][Loki operator].
 
 ```
-juju relate alertmanager-k8s prometheus-k8s
+juju relate alertmanager-k8s:alerting prometheus-k8s:alerting
+```
+
+### Karma dashboard
+
+```yaml
+  karma-dashboard:
+    interface: karma_dashboard
 ```
 
-### karma_dashboard
 The [`karma_dashboard`](https://charmhub.io/karma-k8s/libraries/karma_dashboard)
 relation interface links an entire Alertmanager cluster to a
-[Karma][Karma operator] dashboard.
+[Karma](https://charmhub.io/karma-k8s) dashboard.
 Scaling alertmanager would automatically cause karma to group alerts by
 cluster.
 
 ```
-juju relate alertmanager-k8s karma-k8s
+juju relate alertmanager-k8s:karma_dashboard karma-k8s:karma_dashboard
+```
+
+### Self metrics endpoint
+
+
+```yaml
+self-metrics-endpoint:
+    interface: prometheus_scrape
+```
+This Alertmanager charm may forward information about its metrics endpoint and associated alert rules to a Prometheus charm over the `self-metrics-endpoint` relation using the [`prometheus_scrape`](https://charmhub.io/prometheus-k8s/libraries/prometheus_scrape) interface. In order for these metrics to be aggregated by the remote Prometheus charm all that is required is to relate the two charms as in:
+
+```bash
+juju relate alertmanager-k8s:self-metrics-endpoint prometheus:metrics-endpoint
+```
+
+
+### Grafana dashboard
+
+```yaml
+  grafana-dashboard:
+    interface: grafana_dashboard
+```
+
+Over the `grafana-dashboard` relation using the [`grafana-dashboard`](https://charmhub.io/grafana-k8s/libraries/grafana_dashboard) interface, this Alertmanager charm also provides meaningful dashboards about its metrics to be shown in a [Grafana Charm ](https://charmhub.io/grafana-k8s).
+
+In order to add these dashboards to Grafana all that is required is to relate the two charms in the following way:
+
+```bash
+juju relate alertmanager-k8s:grafana-dashboard grafana-k8s:grafana-dashboard
+```
+
+### Grafana Source
```

Comment:
To be consistent with the other header, only the first word should be capitalized (which I like!)
```suggestion
### Grafana source
```